crypto/store/store_lib.c

```diff
@@ -1095,3 +1095,62 @@ OSSL_STORE_CTX *OSSL_STORE_attach(BIO *bp, const char *scheme,
 
     return ctx;
 }
+
+const char *OSSL_file_stat(const char *uri, struct stat *st)
+{
+    const char *path = uri, *q;
+    struct stat local_st;
+
+    if (st == NULL)
+        st = &local_st;
+
+    /*
+     * First, unless the URI starts with "file://",
+     * try and see if the full URI can be taken as a local file path name.
+     */
+    if (!HAS_CASE_PREFIX(uri, "file://")) {
+        if (stat(path, st) == 0)
+            return uri;
+        ERR_raise_data(ERR_LIB_SYS, errno, "calling stat(%s)", path);
+    }
```

Comment:
This is
1. not quite right, you ignore the possible authority, where we do accept `localhost` (see below)
2. unnecessary, exactly because the check is already present below.

Reply:
Please note that I aim at faithfully following the semantics that you had implemented originally,
while simplifying the code: unfolding the partly degenerate loop and removing the internal temporary list of URIs.

The extra check done here: `!HAS_CASE_PREFIX(uri, "file://")` is needed for this reason,
because you covered the "file://" case (via two conditions on "file:" and "//") later, invalidating the first entry in the list.
The new code has the same effect, but does so by simpler means.

Reply:
No, the way I coded it in this PR just stats `/home/levitte/.bashrc`, like the original code does.

You can try it out:
```
apps/openssl verify -CAstore file://localhost/home/levitte/.bashrc -CAfile test/certs/root-cert.pem test/certs/root-cert.pem
```
now yields
```
Error loading store URI file://localhost/home/levitte/.bashrc
80CC5EF401000000:error:80000002:system library:OSSL_file_stat:No such file or directory:crypto/store/store_lib.c:1154:calling stat(/home/levitte/.bashrc)
```
which is exactly what is wanted/needed.

Reply:
Never mind this discussion, got it now